### DISCUSSION TITLE
[Roadmap] 废弃 PJAX，使用 Valaxy 构建主题新版本

### DISCUSSION THREAD

**Discussion by YunYouJun** [YunYouJun/hexo-theme-starry#260] [General 大局为重]
### [Valaxy](https://github.com/YunYouJun/valaxy) 与 [valaxy-theme-yun](https://github.com/YunYouJun/valaxy/tree/main/packages/valaxy-theme-yun) 已基本可用！

但毫无疑问，它可能存在很多细节问题。在足够稳定后，它将发布 v1.0 版本。
也欢迎任何的试用与反馈！

> 只需一行命令即可快速创建启动示例项目！

```bash
// 若没有 pnpm，执行 npm i -g pnpm
pnpm create valaxy
```

### Try it Online

[![StackBlitz](https://developer.stackblitz.com/img/open_in_stackblitz.svg)](https://stackblitz.com/edit/valaxy)

> 更多信息请参见 [Valaxy 文档](https://valaxy.site)，正在不断更新中。
---

## 目前问题

主题目前大部分 BUG 均来自 PJAX 问题，而这一定程度上影响了主题的开发调试效率。

同时 Hexo 本身的基础架构和功能已逐渐落后，譬如文档和类型提示等做的并不好，无法使用 Vite 等 ESM 极速与热更新，无法利用 React/Vue 的生态和构建体验（指非 CDN 方式），同样降低了主题开发体验。

此外，PJAX 实际上最早与 jQuery 紧密相连，譬如 [jquery-pjax](https://github.com/defunkt/jquery-pjax)，但随着 jQuery 逐渐退出历史舞台，jquery-pjax 几年前便几乎不再维护。
尽管也有原生 JS 版本的 [pjax](https://github.com/MoOx/pjax)，但同样早已在几年前几乎不再更新。

而这究其原因，一定程度上由于现在前端开发框架（Vue/React）的攻城略地而导致，用户不再需要费心费力去调试 PJAX，局部更新页面内容，现代前端框架几乎都已经完成了该项工作，而仅需简单配置。

- jsdelivr 在国内的陨落也导致广大站长为 CDN 发起了愁，这也意味着我们不得不尽可能减少公共 CDN 的使用，转为使用自行引入构建的方式发布。 

## 计划

因此，本主题计划在 v1.9.x 版本废弃 PJAX，仅支持普通静态页面，并进入维护模式。

由于 Hexo 历史性的原因，存在诸多不便。
我决定从源头构建新一代静态博客框架已解决这个问题。并将开发默认主题 `valaxy-theme-yun`，实现现有主题风格与功能。

现已开源，活跃开发中。见 [valaxy](https://github.com/YunYouJun/valaxy)。

它的特性包括但不局限于：

- 完全局部热更新、与配置热加载
- Vite 极速开发体验
- 任意使用 [icones.js.org](https://icones.js.org/) 中所有图标，而无需配置
- 内置的类 TailwindCSS 工具类，且按需加载
- 在文章中插入 `HTML` 或 `Vue` 组件
- 配置类型提示
- ...

## 其它

如果您有其它想法或建议，欢迎留言。

**Comment by eallion** [comment]
希望能看到 5k 和 100k 文章数量的构建性能测试结果。不知道局部热更新是否只构建增量内容。

**Reply by YunYouJun** [reply]
在性能上，我更多指的是对于用户的开发和浏览体验。
这方面可以简要对比 https://valaxy.yyj.moe/ 与 https://www.yunyoujun.cn ，我认为它的差别十分显著。

由于 CI 的存在，构建性能差距似乎不是那么重要。
当然，在此之后我会提供一个性能 benchmark 。在构建上，它的速度相比 hexo 同样优秀。

**Comment by YunYouJun** [comment]
当前的一些进展与特色预览：https://github.com/YunYouJun/valaxy/discussions/116

**Comment by misaka-clover** [comment]
支持云游前辈！XD
